### PR TITLE
Secrets docs

### PR DESCRIPTION
**Pull request by TylerMizuyabu**

### CODE REVIEW

File: organizations/SECRETS.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+`selected_repositories` is an optional list of repository id strings for which repositories should have access to the secret. Only set this field if visibility has been set to `selected`. It defaults to an empty list if not set so the secret won't be accessible to any repositories if it's not set.
+
+Either module will work well for organization secret management, but in the event that you want to do something like make the terragrunt configuration managing secrets dependant on the terragrunt configuration you are using to create repositories. But don't want the rest of organization management to wait on repository creation before it can execute. Then you might want to consider creating a `secrets` folder and configuration under the `projects/repositories/<ORG-NAME>` path that uses the organization secrets module to create these secrets. Keep in mind that while spreading organization secret definitions out like this can be helpful to keep context of what the secret is used for and where, it also might lead to naming conflicts which would result in one configuration overwriting the secret created by another configuration.
```
Comment (blastdan) [inline organizations/SECRETS.md:38]: Should we always suggest the secrets folder?
It makes sense for a lot of reasons. Changes inside it would always be a red flag etc.

Comment (TylerMizuyabu) [reply organizations/SECRETS.md:38]: I think suggesting it for secrets with visibility type `selected` makes a lot of sense. Visibility type `all` doesn't really make that much sense, and I'm not sure about private. And the only thing I worry is losing track of what secret names have already been taken resulting in modules overwriting each others secrets

Comment (TylerMizuyabu) [reply organizations/SECRETS.md:38]: I guess `private` would be like `all` and it doesn't make much sense to put it at the project level. But yeah we could probably reword this to something along the lines of

"When creating secrets that you only want to grant a select amount of repositories access you should consider creating a `secrets` folder...."

Then just warn them about overwrites

Comment (blastdan) [reply organizations/SECRETS.md:38]: Oh I think I understand.

I was considering the idea of a "Global" project that we could standardize on.

Something for shared teams, or org secrets.

Comment (TylerMizuyabu) [reply organizations/SECRETS.md:38]: We could also consider a change in how secrets are done. For example, define them all org secrets at the org level. But for visibility of type `select` we set it initially to `[]` then in the repository level we could make use of the tf resources like this https://registry.terraform.io/providers/integrations/github/latest/docs/resources/actions_organization_secret_repositories to grant them access to the secrets. Would avoid potential overwrite

File: organizations/SECRETS.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    encrypted_value       = string
+    visibility            = string
+}
+```
+
+`encrypted_value` is the encrypted value of the secret you wish to store. See [encryption](#encryption) for instructions on how to encrypt your secret.
+
+`visibility` configures what repositories can access your secret. It must be one of `all`, `private`, or `selected`.
+
+If visibility is set to `selected` the secret will be created however no repositories will have access to it. To grant access to the secret make use of the repository set's `public_repositories` and `private_repositories` fields. Each of the fields are a map of object that contain the following optional fields:
```
Comment (bzarboni1) [inline organizations/SECRETS.md:33]:
```suggestion
If visibility is set to `selected` the secret will be created however no repositories will have access to it. To grant access to the secret make use of the `repository_set`'s `public_repositories` and `private_repositories` fields. Each of the fields are a map of object that contain the following optional fields:
```

File: organizations/SECRETS.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+## Usage
+
+### Encryption
+
+Before using any of the modules to create your github secrets it is important to encrypt them first. Github has documentation outlining how to encrypt secrets for their rest API that can be found here: https://docs.github.com/en/rest/guides/encrypting-secrets-for-the-rest-api
+
+### Organization-level secrets
+
+The organiztion and organization secrets modules have the following optional fields for organization-level secret management:
```
Comment (bzarboni1) [inline organizations/SECRETS.md:15]:
```suggestion
The organization and organization secrets modules have the following optional fields for organization-level secret management:
```

File: organizations/SECRETS.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+`encrypted_value` is the encrypted value of the secret you wish to store. See [encryption](#encryption) for instructions on how to encrypt your secret.
+
+`visibility` configures what repositories can access your secret. It must be one of `all`, `private`, or `selected`.
+
+If visibility is set to `selected` the secret will be created however no repositories will have access to it. To grant access to the secret make use of the repository set's `public_repositories` and `private_repositories` fields. Each of the fields are a map of object that contain the following optional fields:
+
+- `organization_action_secrets`: for granting access to organization-level action secrets
+- `organization_codespace_secrets`: for granting access to organization-level codespace secrets
+- `organization_dependabot_secrets` : for granting access to organization-level dependabot secrets
```
Comment (bzarboni1) [inline organizations/SECRETS.md:37]:
```suggestion
- `organization_action_secrets`: for granting access to organization-level Action secrets
- `organization_codespace_secrets`: for granting access to organization-level Codespace secrets
- `organization_dependabot_secrets` : for granting access to organization-level Dependabot secrets
```

File: organizations/SECRETS.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+Before using any of the modules to create your github secrets it is important to encrypt them first. Github has documentation outlining how to encrypt secrets for their rest API that can be found here: https://docs.github.com/en/rest/guides/encrypting-secrets-for-the-rest-api
+
+### Organization-level secrets
+
+The organiztion and organization secrets modules have the following optional fields for organization-level secret management:
+
+- `actions_secrets`: for creating and managing organization secrets accessible to github actions
+- `codespaces_secrets`: for creating and managing organization secrets accessible to github codespaces
+- `dependabot_secrets`: for creating and managing organization secrets accessible to dependabot
```
Comment (bzarboni1) [inline organizations/SECRETS.md:19]:
```suggestion
- `actions_secrets`: for creating and managing organization secrets accessible to GitHub Actions
- `codespaces_secrets`: for creating and managing organization secrets accessible to GitHub Codespaces
- `dependabot_secrets`: for creating and managing organization secrets accessible to Dependabot
```

File: organizations/SECRETS.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+      organization_action_secrets     = ["SECRET_ONE", "SECRET_TWO"] 
+      organization_dependabot_secrets = ["SECRET_FOUR"]
+    }
+  }
+}
+```
+
+### Repository-level secrets
+
+The repository set module's `private_repositories` and `public_repositories` fields accepts the following optional fields for repository secret management:
```
Comment (bzarboni1) [inline organizations/SECRETS.md:83]:
```suggestion
The `repository_set` module's `private_repositories` and `public_repositories` fields accepts the following optional fields for repository secret management:
```

**Review by blastdan — APPROVED** [review]

File: organizations/SECRETS.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+- `action_secrets`: for creating repository secrets accessible to github actions
+- `codespace_secrets`: for creating and managing repository secrets accessible to github codespaces
+- `dependabot_secrets`: for creating and managing repository secrets accessible to dependabot
+- `environments`: for creating and managing environments, including their action secrets.
```
Comment (bzarboni1) [inline organizations/SECRETS.md:87]:
```suggestion
- `action_secrets`: for creating repository secrets accessible to GitHub Actions
- `codespace_secrets`: for creating and managing repository secrets accessible to GitHub Codespaces
- `dependabot_secrets`: for creating and managing repository secrets accessible to Dependabot
- `environments`: for creating and managing environments, including their action secrets.
```

**Review by bzarboni1 — APPROVED** [review]
